core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -174,15 +162,43 @@ oppia.directive('audioControls', [
               AudioPlayerService.getCurrentAudioControlsDirectiveId();
           };
 
+          var onFinishedLoadingAudio = function(loadedFilename) {
+            var audioTranslation = getAudioTranslationInCurrentLanguage();
+            if (requestedToPlayAudio && loadedFilename ===
+                audioTranslation.filename) {
+              requestedToPlayAudio = false;
+              AudioPlayerService.load(audioTranslation.filename, directiveId)
+                .then(function() {
+                  AudioPlayerService.play();
+                });
+            }
+            $scope.audioIsLoading = false;
+          };
+
+          if (!AudioPreloaderService.isRunning()) {
```

Comment:
Just to check, is this meant to be in a function somewhere, or is it meant to be called once at the outset on init?

Reply:
It's meant to be just called once at the outset. I had to put this here instead of ExplorationPlayerService.init because I needed to pass in the callback. I feel like there should be a cleaner way of doing this but I'm not sure how.

Reply:
Ah ok. Do my comments [here](https://github.com/oppia/oppia/pull/4149#discussion_r154566537) help?

Reply:
Moved into ExplorationPlayerService.init